tests/services/test_text_processor.py

```
@@ -161,22 +154,20 @@ def test_context_manager(self) -> None:
     def test_context_manager_cleanup_on_exception(self) -> None:
         """Context manager calls cleanup even when an exception occurs."""
         mock_model = MagicMock()
         mock_model.is_initialized = False
         with patch("services.text_processor.get_text_model", return_value=mock_model):
-            with patch("services.text_processor.ensure_nltk_data"):
-                with pytest.raises(RuntimeError, match="boom"):
-                    with TextProcessor() as _processor:
-                        raise RuntimeError("boom")
-                mock_model.safe_cleanup.assert_called_once()
+            with pytest.raises(RuntimeError, match="boom"):
+                with TextProcessor() as _processor:
```

Ruff (0.15.10)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)